frontend/src/pages/Register.js

```diff
@@ -0,0 +1,201 @@
+  height: 40px;
+  background-color: #d5f5f2;
+  border: none;
+  cursor: pointer;
+  font-size: 24px;
+  padding: 15px;
+  border-radius: 10px;
+  font-family: "Josefin Sans", sans-serif;
+  box-shadow: 5px 5px 10px #888888;
+`;
+
+const Register = () => {
+  const [membername, setMembername] = useState("");
+  const [password, setPassword] = useState("");
+  const [email, setEmailAddress] = useState("");
+  const [location, setLocation] = useState("");
+  const [status, setStatus] = useState("");
+  const [loading, setLoading] = useState(false)
+
+  const accessToken = useSelector((store) => store.member.accessToken);
+  const errors = useSelector((store) => store.member.error);
+  
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (accessToken) {
+      swal('Welcome Thek-Friend! Thank you for becomming a member, you wonderful person!', {icon: 'success', button: 'ok'})
+      navigate("/signin");
+    }
+  }, [accessToken, navigate]);
+
+  
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ membername, password, email, location, status }),
+    };
+    setLoading(true)  
+    fetch(API_URL("signup"), options)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.success) {
+          batch(() => {
+            dispatch(member.actions.setMemberId(data.response.memberId));
+            dispatch(member.actions.setMembername(data.response.membername));
+            dispatch(member.actions.setAccessToken(data.response.accessToken));
+            dispatch(member.actions.setEmailAddress(data.response.email));
+            dispatch(member.actions.setLocation(data.response.location));
+            dispatch(member.actions.setStatus(data.response.status));
+            dispatch(member.actions.setError(null));
+          });
+        } else {
+          batch(() => {
+            dispatch(member.actions.setMemberId(null));
+            dispatch(member.actions.setMembername(null));
+            dispatch(member.actions.setAccessToken(null));
+            dispatch(member.actions.setEmailAddress(null));
+            dispatch(member.actions.setLocation(null));
+            dispatch(member.actions.setStatus(null));
+            dispatch(member.actions.setError(data.response));
+          });
+        }
+      }).finally(() => setLoading(false));
+  };
+
+  return (
+    <>
+   <Box>
+     <Menu/>
+    {loading && <Loader/>}
+      <Form onSubmit={onFormSubmit}>
+
+      <label htmlFor="membernameInput"></label>
+        <Input
+          id="membernameInput"
+          type="text"
+          value={membername}
+          onChange={(e) => setMembername(e.target.value)}
+          minLength="5"
+          required
+          placeholder="User name"
+        />
+       
+       <label htmlFor="emailInput"></label>
```

Comment:
There is not reason to use labels if they are empty - their main purpose is to display text about corresponding input

Reply:
Hi Maks, Thank you for the code review. I am presently working on your suggestions but just thought I would ask about this one......I added the label because I wanted to check the HTML and CSS for accessibility and there were many red-flags when I used the WAVE Chrome Extension for Inputs without Labels..... How should I resolve this? I suppose in theory I should use a label instead of a placeholder, but I prefer the aesthetic of a placeholder ;-)

Reply:
Hi Kara! Definitely labels are more accessible, but I believe you go for placeholders if you prefer them :)
However, if you would decide to choose label approach, they would need to be non empty then, as in example below :

```
<label htmlFor="emailInput">Your email</label>
<Input
  id="emailInput"
  type="email"
  value={email}
  onChange={(e) => setEmailAddress(e.target.value)}
  required
  placeholder="Email"
/>
```